src/repl/scala/tools/nsc/interpreter/IMain.scala

```diff
@@ -105,6 +105,29 @@ class IMain(initialSettings: Settings, protected val out: JPrintWriter) extends
     try body
     finally if (!saved) settings.nowarn.value = false
   }
+  def withSuppressedSettings[A](body: => A): A = {
```

Comment:
At first I was just lazy; the other diff is that we want warnings (?) in one case but not the other. Probably I was just lazy.